pkg/egresspolicy/manager.go

```diff
@@ -0,0 +1,461 @@
+				// identified by pod and config
+				if podData, ok := manager.podDataStore[podId]; ok {
+					err := manager.deleteEgressMap(storedConfig, podData)
+					if err != nil {
+						return err
+					}
+				} else {
+					return fmt.Errorf("Failed to get pod data for %v", podId)
+				}
+			} else {
+				newPolicyList = append(newPolicyList, policyId)
+			}
+		}
+		if len(newPolicyList) > 0 {
+			manager.policyPods[podId] = newPolicyList
+		} else {
+			// podDataStore untouched here since pod data is unchanged
+			delete(manager.policyPods, podId)
+		}
+	}
+	delete(manager.policyConfigs, config.id)
+	return nil
+}
+
+// OnAddPod is the event handler for pod additions.
+func (manager *Manager) OnAddPod(pod *slim_corev1.Pod) {
+	manager.mutex.Lock()
+	defer manager.mutex.Unlock()
+
+	if len(manager.policyConfigs) == 0 {
+		return
+	}
+	// If the pod already exists in the internal cache, ignore all the subsequent
+	// events since they'll be handled in the OnUpdatePod callback.
+	// GH issue #13136
+	id := types.NamespacedName{
+		Name:      pod.GetName(),
+		Namespace: pod.GetNamespace(),
+	}
+	if _, ok := manager.policyPods[id]; ok {
+		return
+	}
+	manager.OnUpdatePodLocked(pod, false, true)
+}
+
+// OnUpdatePodLocked is the handler for pod info updates.
+func (manager *Manager) OnUpdatePodLocked(pod *slim_corev1.Pod, removeOld bool, upsertNew bool) {
+	var podData *podMetadata
+	var err error
+
+	if len(manager.policyConfigs) == 0 {
+		return
+	}
+
+	podIPs := k8sUtils.ValidIPs(pod.Status)
+	if len(podIPs) == 0 {
+		log.WithError(err).WithFields(logrus.Fields{
+			logfields.K8sPodName:   pod.Name,
+			logfields.K8sNamespace: pod.Namespace,
+		}).Error("Failed to get valid pod ips, skipping update to Egress Policy")
+		return
+	}
+
+	if podData, err = manager.getPodMetadata(pod, podIPs); err != nil {
+		log.WithError(err).WithFields(logrus.Fields{
+			logfields.K8sPodName:   pod.Name,
+			logfields.K8sNamespace: pod.Namespace,
+		}).Error("Failed to get valid pod metadata, skipping update to Egress Policy")
+		return
+	}
+
+	if removeOld {
+		// Check if the pod was previously selected by any of the policies.
+		if policies, ok := manager.policyPods[podData.id]; ok {
+			for _, policy := range policies {
+				config := manager.policyConfigs[policy]
+				err := manager.deleteEgressMap(config, podData)
+				if err != nil {
+					log.WithError(err).Errorf("Error updating pod mapping for pod %v", podData.id)
+					return
+				}
+			}
+			delete(manager.policyPods, podData.id)
+			delete(manager.podDataStore, podData.id)
+		}
+	}
+
+	if upsertNew {
+		// Check if any of the current policies select this pod.
+		for _, config := range manager.policyConfigs {
+			if config.policyConfigSelectsPod(podData) {
+				err := manager.upsertPolicyPod(config, podData)
+				if err != nil {
+					log.WithError(err).Errorf("Error upserting pod mapping for pod %v", podData.id)
+					return
+				}
+			}
+		}
+	}
+}
+
+//OnUpdatePod is the event handler for pod updates.
```

Comment:
Nit: 
```suggestion
// OnUpdatePod is the event handler for pod updates.
```

Please fix any occurrences of these

Reply:
Fixed.